src/utils/symlink.ts

```diff
@@ -39,29 +45,80 @@ export function createSymlinks(
-    );
-  } else {
     logger.debug(
-      `Changing symlink for "latest.json" from version "${oldVersion}" to "${newVersion}"`
+      `${parsedOldVersion ? 'Updating' : 'Adding'} symlink for "latest.json" ${
+        parsedOldVersion ? `from ${oldVersion} ` : ''
+      }to "${newVersion}"`
     );
     forceSymlink(baseVersionName, path.join(packageDir, 'latest.json'));
   }
 
-  // link major
-  const majorVersionLink = `${parsedNewVersion.major}.json`;
-  forceSymlink(baseVersionName, path.join(packageDir, majorVersionLink));
+  // Read possibly existing symlinks for major and minor versions of the new version
+  const newVersionMajorSymlinkedVersion = getExistingSymlinkedVersion(
+    path.join(packageDir, `${parsedNewVersion.major}.json`)
+  );
+  const newVersionMinorSymlinkedVersion = getExistingSymlinkedVersion(
+    path.join(
+      packageDir,
+      `${parsedNewVersion.major}.${parsedNewVersion.minor}.json`
+    )
+  );
+
+  // link {major}.json if there's no link yet for that major
+  // or if the new version is newer than the currently linked one
+  if (
+    !newVersionMajorSymlinkedVersion ||
+    versionGreaterOrEqualThan(parsedNewVersion, newVersionMajorSymlinkedVersion)
+  ) {
+    const majorVersionLink = `${parsedNewVersion.major}.json`;
+    logger.debug(
+      `${
+        newVersionMajorSymlinkedVersion ? 'Updating' : 'Adding'
+      } symlink for "${majorVersionLink}" ${
+        newVersionMajorSymlinkedVersion
+          ? `from version "${semVerToString(newVersionMajorSymlinkedVersion)}" `
+          : ''
+      }to "${newVersion}"`
+    );
+    forceSymlink(baseVersionName, path.join(packageDir, majorVersionLink));
+  }
+
+  // link {minor}.json if there's no link yet for that minor
+  // or if the new version is newer than the currently linked one
+  if (
+    !newVersionMinorSymlinkedVersion ||
+    versionGreaterOrEqualThan(parsedNewVersion, newVersionMinorSymlinkedVersion)
+  ) {
+    const minorVersionLink = `${parsedNewVersion.major}.${parsedNewVersion.minor}.json`;
+    logger.debug(
+      `${
+        newVersionMinorSymlinkedVersion ? 'Updating' : 'Adding'
+      } symlink for "${minorVersionLink}" ${
+        newVersionMinorSymlinkedVersion
+          ? `from version "${semVerToString(newVersionMinorSymlinkedVersion)}" `
+          : ''
+      }to "${newVersion}"`
+    );
+    forceSymlink(baseVersionName, path.join(packageDir, minorVersionLink));
+  }
+}
 
-  // link minor
-  const minorVersionLink = `${parsedNewVersion.major}.${parsedNewVersion.minor}.json`;
-  forceSymlink(baseVersionName, path.join(packageDir, minorVersionLink));
+function getExistingSymlinkedVersion(symlinkPath: string): SemVer | null {
+  try {
+    // using lstat instead of exists because broken symlinks return false for exists
+    fs.lstatSync(symlinkPath);
+    const linkedFile = fs.readlinkSync(symlinkPath);
+    return parseVersion(path.basename(linkedFile));
+  } catch {
```

Comment:
this try catch is overly broad -- each of these 3 statements could raise errors but I think (?) you only intend to catch errors from the first one?  this could hide a programming error especially after future maintenance

Reply:
good catch!